forte/processors/base/data_augment_processor.py

```diff
@@ -46,44 +55,559 @@ class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
+            to a dictionary (position -> length) inserted
+            by :func: insert.
+            The position is the index in the original datapack
+            of insertion, and the length is the length of the
+            inserted string.
+
+        - deleted_annos: {datapack id: Set[annotation tid]}
+            It records the deleted spans, mapping from datapack id
+            to a set of annotation tids appended by :func: delete.
+
+        - data_pack_map: {orig pack id: new pack id}
+            It maintains a mapping from the pack id
+            of the original pack to the pack id of augmented pack.
+            It is used when copying the MultiPackLink.
+
+        - anno_maps: {datapack id: Dict{orig tid, new tid}}
+            It is a map for tracking the annotation ids
+            before and after the auto align. It maps the
+            original annotation tid to the new annotation tid.
+            It is used when copying the MultiPackLink.
         """
         super().__init__()
-        self.replaced_spans: List[Tuple[Entry, str]] = []
+        self.replaced_annos: DefaultDict[int, List[Tuple[Span, str]]] = \
+            defaultdict(list)
+        self.inserted_annos: DefaultDict[int, Dict[int, int]] = \
+            defaultdict(dict)
+        self.deleted_annos: DefaultDict[int, Set[int]] = defaultdict(set)
+
+        self.data_pack_map: Dict[int, int] = {}
+        self.anno_maps: Dict[int, Dict[int, int]] = {}
+        self.other_entry_policy: Dict[str, str] = {}
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        # Parse the "other_entry_policy" in the configuration.
+        self.other_entry_policy = self._get_other_entry_policy_dict()
 
-    def replace(self, replacement_op: TextReplacementOp, input: Entry):
+    def _get_other_entry_policy_dict(self) -> Dict[str, str]:
+        r"""
+        A helper function for parsing the "other_entry_policy"
+        configuration.
         """
+        entries = self.configs['other_entry_policy']["entry"]
+        policies = self.configs['other_entry_policy']["policy"]
+        other_entry_policy = {}
+        for entry, policy in zip(entries, policies):
+            other_entry_policy[entry] = policy
+        return other_entry_policy
+
+    def _is_span_overlap(self, pid: int, begin: int, end: int) -> bool:
+        r"""
+        This function will check whether the new span
+        has an overlap with any existing spans.
+        Args:
+            - pid: Datapack Id.
+            - begin: The span begin index.
+            - end: The span end index.
+        Returns:
+            True if the input span overlaps with
+            any existing spans, False otherwise.
+        """
+        for span, _ in self.replaced_annos[pid]:
+            if not(span.begin >= end or span.end <= begin):
+                return True
+        return False
+
+    def replace(
```

Comment:
Please do not expose these internal functions to users, rename them like `__replace`